integration_tests/src/main/python/cache_test.py

```diff
@@ -212,3 +212,30 @@ def test_cache_partial_load(data_gen, enableVectorizedConf):
             .cache()
             .limit(50).select(f.col("b")), enableVectorizedConf
     )
+
+@pytest.mark.parametrize('data_gen', all_gen, ids=idfn)
+@pytest.mark.parametrize('ts_write', ['INT96', 'TIMESTAMP_MICROS', 'TIMESTAMP_MILLIS'])
+@pytest.mark.parametrize('enableVectorized', ['true', 'false'], ids=idfn)
+@allow_non_gpu('CollectLimitExec', 'DataWritingCommandExec')
+def test_cache_columnar(spark_tmp_path, data_gen, enableVectorized, ts_write):
+    data_path_cpu = spark_tmp_path + '/PARQUET_DATA_CPU'
+    data_path_gpu = spark_tmp_path + '/PARQUET_DATA_GPU'
+    def read_parquet_cached(data_path):
+        def write_read_parquet_cached(spark):
+            df = unary_op_df(spark, data_gen).select(f.col("a"))
+            df.write.parquet(data_path)
+            cached = spark.read.parquet(data_path).cache()
+            cached.count()
+            cached.limit(50)
```

Comment:
I don't think this does anything useful.  It will create a data frame that will limit cached to 50, but then it throws it away and does nothing with it.

Reply:
It tests the path where we write the `ColumnarBatch` to a `CachedBatch`. Aren't we still comparing the two DF's that result from calling the `limit(50)`?

Reply:
No because it is not returned.  The only things that are compared are what is returned by the function.

Reply:
Because this is python if there is no return statement a None is returned.  This means that from_cpu and from_gpu are both None and will always pass.

Reply:
:facepalm:

Reply:
good thing that I tested the code path by setting breakpoints. So I am positive the path is being exercised. Let me add the return statement so the outputs are compared! doh!